pkg/cmd/repo/setdefault/setdefault.go

```diff
@@ -120,10 +120,9 @@ func setDefaultRun(opts *SetDefaultOptions) error {
 
 	if opts.ViewMode {
 		if currentDefaultRepo == nil {
-			fmt.Fprintln(opts.IO.Out, "no default repository has been set; use `gh repo set-default` to select one")
-		} else {
-			fmt.Fprintln(opts.IO.Out, displayRemoteRepoName(currentDefaultRepo))
+			return errors.New("no default repository has been set; use `gh repo set-default` to select one")
```

Comment:
I am not sure I agree that this is an error state for this command. In order to make this command easier to use in scripts, what are your thoughts on skipping the output output when in non-TTY mode and there is no default repository?

Reply:
> what are your thoughts on skipping the output output when in non-TTY mode and there is no default repository?

It's a thing some CLIs do (`git branch --show-current` is the most recent command I used that does this) so it seems fine to me. I do think a non-zero exit code does make a lot of sense here but if you don't think that's the right choice, I'm not going to oppose just returning nothing with a zero exit code since it's still an improvement that allows me to use it in a script. Just say the word and I'll make the changes :)

Reply:
@Jackenmen Yeah lets move the direction I laid out above 👍